examples/http-default-error-handling/http_default_error_handling.md

```diff
@@ -0,0 +1,19 @@
+# HTTP service - Error handling
+
+Error handling is an integral part of any network program. The error can be returned or checked from the resource function, interceptor, request dispatcher, or data binder. These errors are often handled by a default handler and sent back as `500 InternalSeverError` responses with the error message in the body. The error is printed on the server terminal along with the stacktrace. This prevents some additional error-handling codes in your program.
```

Comment:
Error handling is an integral part of any network program. Application logic written inside a resource method could propagate errors from the resource method to the `http:Listener`. Though there are many ways to propagate errors, the most common way is to use the `check` keyword. `http:Listener` intercepts these errors and sends a `500 Internal Server Error` response with the error message in the payload. In addition, it logs the error with the stack-trace in the console. Users can take control of this behavior by adding a `do/fail` block in the resource method and handling the errors themselves. When there is a repetition of error handling logic in resource method, the repetition can be avoided by moving the logic to an Error handler interceptor. 

> Note: Errors originated from the `http:Listener` itself due to resource not found, data-binding, authorization, etc are converted into respective status codes such as `500 Internal Server Error`, `400 Bad Request`, `401 Unauthorized`, etc. Again, users can take control of this behavior by adding an Error handler interceptor.